engine/Default/planet_list.inc

```diff
@@ -0,0 +1,46 @@
+<?php
+
+// The engine files for planet lists have a lot in common, so do
+// most of the work here.
+function planet_list_common($allianceId, $getPlanets) {
+	global $template, $player;
+
+	$playerOnly = $allianceId == 0;
```

Comment:
This should be `!$player->hasAlliance()`, that way the method of checking if a player has an alliance or not is contained in the one place and it also allows removing the check below, because you know they can never both be true.  And in fact the uses of `$playerOnly` could probably be replaced by `hasAlliance` calls to make the meaning clearer in the world of the combined planet lists

Reply:
I would really like to replace `$playerOnly` with `!$player->hasAlliance()`, but that would break existing support for hidden players viewing alliance planet lists (since the hidden player might not be in an alliance). I _believe_ that's the only scenario where you'd have a solo player viewing another alliance's planet list.

https://github.com/smrealms/smr/blob/17d7492baa8f55df33eeeb224117e5c32dab90a7/engine/Default/menu.inc#L49

Reply:
How does that current code here handle the hidden player case? As far as I can see it would treat them as a normal player

Reply:
If the hidden player is not in an alliance, but they want to view the planet list for alliance X (because apparently hidden players are gods), then even though they will have `$player->hasAlliance() == false` and `$player->getAllianceID() == 0`, the page will have `$allianceId == X`. This will allow the planet list for alliance X to display. Otherwise, it would assume we want to only display the current player's planet.

Reply:
Urg, ok, and yeah they are but I've never made use of it (you can't have a hidden player and also play yourself imo)

Reply:
Well, we could get rid of the notion of hidden players, which would certainly simplify something like this. In what context were they used in the past?

Reply:
I don't really know, I think it was mostly Azool and @MichaelKunze that used them.  My guess was always that it was a way to look into problems that only occurred for other alliances or as admin things to check if there was anything suspicious going on.  What I personally did in the rare occasion when someone had a bug unique to them for some reason was to use a "login as" type functionality, where I would change my `account_id` in `active_session` to theirs in order to reproduce - it was a pretty uncommon thing to need though, especially after I added the automatic bug reporting (because the majority of stuff is never reported.. and the automatic report can include a load more info as well)

Reply:
Thats something Azool implemented after me. I never used it. Get rid of it. @Page- 's solution works perfectly and did it that way when really needed.